src/solvers/td3.ml

```diff
@@ -297,10 +308,26 @@ module WP =
         delete_marked wpoint;
         delete_marked stable;
 
-        print_data data "Data after clean-up"
+
+        print_data data "Data after clean-up";
+
+        List.iter set_start st;
+
+        if not !any_changed_start_state && GobConfig.get_bool "incremental.reluctant" then (
+          (* solve on the return node of changed functions. Only destabilize influenced nodes outside the function if the analysis result changed *)
+          print_endline "Separately solving changed functions...";
+          Hashtbl.iter (fun x (old_rho, old_infl) -> ignore @@ Pretty.printf "test for %a\n" Node.pretty_trace (S.Var.node x);
+            solve x Widen;
+            if not (S.Dom.leq (HM.find rho x) old_rho) then (
+              HM.replace infl x old_infl;
+              destabilize x; HM.replace stable x ())
+          ) old_ret;
+
+          print_endline "Final solve..."
+        )
       );
 
-      List.iter set_start st;
+      if not (GobConfig.get_bool "incremental.load") then List.iter set_start st;
```

Comment:
Maybe this could be in the `else` branch of the massive `if` before. It's a bit confusing to read it like this because it makes it look like `set_start` is somehow unnecessary for incremental, while it's actually still done in the incremental case, but simply earlier in the `then` branch.